### ISSUE TITLE
xunit.runner.visualstudio cannot load reporter which references to another version of xunit.runner.utility.dll

### ISSUE THREAD

**Issue by nvborisenko** [xunit/xunit#1864]
On file system I have the following files in folder:
```
MyTests.dll
xunit.*.dll
xunit.runner.visualstudio.testadapter.dll
xunit.runner.utility.net452.dll
```

To execute tests I use the command
```cmd
dotnet vstest MyTests.dll
```

Everything works fine. I am executing tests with disabled shadow copying.

Now I have `ReportPortal.XUnitReporter.dll` (custom reporter) which references to `xunit.runner.utility.net452, Version=2.4.1.0`, and I put it into folder in hope runner will use it as reporter. Tests are running, and it seems runner doesn't use my custom reporter.

While investigating it I figured out the following exception why runner cannot load my custom reporter.
```
System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at Xunit.Runner.VisualStudio.VsTestRunner.GetAvailableRunnerReporters(IEnumerable`1 sources) in C:\Projects\xunit\src\xunit.runner.visualstudio\VsTestRunner.cs:line 792
Loader Exception: System.IO.FileNotFoundException: Could not load file or assembly 'xunit.runner.utility.net452, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c' or one of its dependencies. The system cannot find the file specified.
```

My folder contains `xunit.runner.utility.net452.dll` with higher version. It seems that `ReportPortal.XUnitReporter.dll` should be compiled with exact version of `xunit.runner.utility.net452.dll` which user uses.

This issue is found in scope of https://github.com/xunit/xunit/issues/1681

**Comment by bradwilson** [comment]
Yes, there's no easy way for us to fix this today, which is one reason why we don't support people writing custom runners.

**Comment by nvborisenko** [comment]
Did you mean custom reporters? Not runner.

**Comment by bradwilson** [comment]
That is correct.

**Comment by bradwilson** [comment]
Closing this a By Design.

**Comment by nvborisenko** [comment]
Oh nooo, xunit doesn't support custom reporters :( I'm trying to implement it to support [Report Portal](http://reportportal.io) and, based on your position, it will be never implemented.

**Comment by bradwilson** [comment]
With the current design (hard links to xunit.runner.utility) it's not realistic to be able to support it. There is a possibility we could add some abstraction layer for 3.0 for this, but honestly this is the first time it's come up in the several years since we shipped 2.0.
